sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,755 @@
+associated with it, with the allocation kind `usm::host` and applying any
+properties in `props`.
+
+[source, c++]
+----
+template <typename Properties = empty_properties_t>
+memory_pool(context ctx, device dev, usm::alloc kind, Properties props = {});
+----
+
+_Effects_: Constructs a memory pool associated with `ctx` and `dev`, with the
+allocation kind `kind` and applying any properties in `props`.
+
+_Throws_: An exception with the `errc::invalid` error code if `kind` is
+`usm::host`.
+
+[source, c++]
+----
+~memory_pool();
+----
+
+_Effects_: If this was the last copy, signals to the SYCL runtime for the memory
+pool to be destroyed after all remaining allocations have been freed, and
+returns immediately without waiting.
+
+[source, c++]
+----
+context get_context() const;
+----
+
+_Returns_: The SYCL context associated with the memory pool.
+
+[source, c++]
+----
+std::vector<device> get_devices() const;
+----
+
+_Returns_: The SYCL device(s) associated with the memory pool.
+
+[source, c++]
+----
+usm::alloc get_alloc_kind() const;
+----
+
+_Returns_: The memory allocation kind of the memory pool.
+
+[source, c++]
+----
+size_t get_max_size() const;
+----
+
+_Returns_: The maximum size of the memory pool.
+
+[source, c++]
+----
+size_t get_threshold() const;
+----
+
+_Returns_: The deallocation threshold of the memory pool.
+
+[source, c++]
+----
+void set_new_threshold(size_t newThreshold);
+----
+
+_Effects_: Sets the deallocation threshold of the memory pool if the value of
+`newThreshold` is larger than the current threshold.
+
+_Throws_: An exception with the `errc::invalid` error code if the value of
+`newThreshold` is lower than the current threshold or larger than the maximum
+allocation size.
+
+
+=== Memory pool properties
+
+A memory pool can be constructed with a number of properties which can change
+certain behaviors, these can be specified when constructing a `memory_pool`
+object.
+
+[source,c++]
+----
+namespace ext::oneapi::experimental {
+
+struct initial_threshold_key {
+  using value_t = property_value<initial_threshold_key>;
+
+  initial_threshold_key(size_t initialThreshold);
+};
+
+struct maximum_size_key {
+  using value_t = property_value<maximum_size_key>;
+
+  initial_threshold_key(size_t maxSize);
+};
+
+struct read_only_key {
+  using value_t = property_value<read_only_key>;
+
+  read_only_key(bool readOnly);
+};
+
+struct zero_init_key {
+  using value_t = property_value<zero_init_key>;
+
+  read_only_key(bool zeroInit);
+};
+
+struct use_existing_memory_key {
+  using value_t = property_value<use_existing_memory_key>;
+
+  use_existing_memory(void *ptr, size_t size);
+};
+
+inline constexpr initial_threshold_key::value_t initial_threshold;
+inline constexpr maximum_size_key::value_t maximum_size;
+inline constexpr read_only_key::value_t read_only;
+inline constexpr zero_init_key::value_t zero_init;
+inline constexpr use_existing_memory_key::value_t use_existing_memory;
```

Comment:
These `inline constexpr` variables are not correct for a runtime property.  See [Representation of Properties](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc#82-representation-of-properties) in the properties spec.

Reply:
Thanks, I've fixed those in https://github.com/intel/llvm/commit/48493ace0babceba6b65fe7399b43284c10681db.